### PR TITLE
readmeeeee

### PR DESCRIPTION
**Pull request by davemcorwin**
Polish up the readme #22

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,79 +1,304 @@
...
+* [Quick Start](#quick-start)
+* [Details](#details)
+* [FAQ](#faq)
+
+## Motivation
+Whenever your application manages users, you will inevitably need to handle email confirmation, password reset, user invitations, and other authentication flows. While not too complicated, they are sort of annoying to implement and some great libraries have our backs. [Devise][devise] and [Sorcery][sorcery] are great options that we have used in the past, but we found ourselves wanting both a little less and a little more.
+
+### Devise
+[Devise][devise] is an amazing gem! It is perfect when you want an all-in-one solution that handles user authentication and associated flows for your Rails/ERB app. Everything is in the box, including the routes, controllers, views, and even mailers to handle user auth. But we often use Rails as an API and/or wanted more control over all those pieces and it became difficult to peel back all the layers to just to confirm a users email.
+
```
Comment (inveterateliterate) [inline README.md]: a user's > I think we need the possessive apostrophe here.

File: README.md [hunk 2]
```diff
@@ -1,79 +1,304 @@
...
+* 'sent'
+* 'completed'
+* 'expired'
+
+`force_confirm!` - forcibly completes the given *tokenable*
+
+See the [Api Docs][docs] for more details.
+
+## Advanced
+Sometimes in order to redeem a token, we want to make sure some additional information is present and possible save that to our model. For example, when implementing a password reset flow, we want to update the User with the new password and make sure that its valid.
```
Comment (inveterateliterate) [inline README.md]: possibly instead of possible

File: README.md [hunk 3]
```diff
@@ -1,79 +1,304 @@
...
+No routing, views, controllers, or mailers, just logic that you can use wherever and whenever you want.
+
+### Authentication strategy agnostic
+Token Master does not handle user authentication, it assumes you have this covered with `has_secure_password`, Devise, Sorcery, or other solutions
+
+### Unobtrusive
+Does not take over your app, minimal magic, and only if you want it. Token Master works with your existing authentication solution.
+
+### Flexible
+Works for Apis, ERB apps and everything in between.
```
Comment (wade00) [inline README.md]: APIs?

File: README.md (hunk 1 above)
Comment (davemcorwin) [reply README.md]: done

File: README.md (hunk 2 above)
Comment (davemcorwin) [reply README.md]: done

File: README.md [hunk 4]
```diff
@@ -1,79 +1,304 @@
...
 
-## Same for foobar
+end
+```
 
-User.foobar_by_token!(token)
+## Details
 
-...
+Lets revisit the Quick Start and fill in the details.
```
Comment (wade00) [inline README.md]: I think it should be "Let's".

File: README.md [hunk 5]
```diff
@@ -1,79 +1,304 @@
...
+```
+token_master :confirm, :invite, :reset, ...
+```
 
-- Token Lifetime (`:token_lifetime`, takes an integer
-- Reuired Params (`:token_lifetime`), takes an array
-- Token Length(`:token_length`), takes an integer
+There are 2 tiny bits of magic here:
 
+1. In Rails apps by default, the Token Master module is included in your `ApplicationRecord` base class. However, if necessary, you can add this yourself by include the following in your class:
```
Comment (wade00) [inline README.md]: "by _including_ the following".

**Review by wade00 — APPROVED** [review]
I can definitely get by with this. I don't think I'd have any trouble getting set up.

The only feedback I have is that I stumbled over "tokenable" a few times. I'm used to hearing it around the office, but calling the action of confirming an account as a "tokenable" feels weird.

**Comment by wade00** [comment]
@davemcorwin I "approved" the review even though I had a few comments. See above!

**Comment by davemcorwin** [comment]
ty @wade00!  I tried to introduce the concept of a *tokenable* but it probably needs more work

**Review by inveterateliterate — APPROVED** [review]

**Pull request merged**